lib/hocon/impl/simple_config.rb

```diff
@@ -107,6 +107,57 @@ def hash
     41 * @object.hash
   end
 
+  def self.find_key_or_null(me, key, expected, original_path)
+    v = me.peek_assuming_resolved(key, original_path)
+
+    if v.nil?
+      raise Hocon::ConfigError::ConfigMissingError.new(nil, original_path.render, nil)
+    end
+
+    if not expected.nil?
```

Comment:
I don't think we've been very consistent about it unfortunately (I've been using unless, but others have been using if not), so I'm fine with leaving this.